GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -658,22 +658,27 @@ private int FindRevisionIndex(ObjectId? objectId)
         /// Returns whether the required revision was found and selected.
         /// </summary>
         /// <param name="objectId">Id of the revision to select.</param>
+        /// <param name="toggleSelection">Toggle if the selected state for the revision.</param>
+        /// <param name="showNoRevisionMsg">Show popup if revision is not visible in grid. Should only be set if the ObjectId is in the repo.</param>
         /// <returns><c>true</c> if the required revision was found and selected, otherwise <c>false</c>.</returns>
-        public bool SetSelectedRevision(ObjectId? objectId, bool toggleSelection = false)
+        public bool SetSelectedRevision(ObjectId? objectId, bool toggleSelection = false, bool showNoRevisionMsg = true)
         {
             var index = FindRevisionIndex(objectId);
 
-            if (index >= 0 && index < _gridView.RowCount)
+            if (index < 0 || index >= _gridView.RowCount)
             {
-                Validates.NotNull(objectId);
-                SetSelectedIndex(index, toggleSelection);
-                _navigationHistory.Push(objectId);
-                return true;
+                if (showNoRevisionMsg)
+                {
+                    MessageBoxes.RevisionFilteredInGrid(this, objectId);
+                }
```

Comment:
I am very much against having the grid show messages, it doesn't feel appropriate.
This API returns `false` if we couldn't find the requested objectid. The caller (i.e., `FormBrowse`) must then show the error message, if necessary (in the interactive flow).